kubeadm: remove some CNI plugins from the kubeadm setup guide

neolit123 opened the pull request:
The kubeadm survey for 1.13:
https://docs.google.com/forms/d/1A855D054FZfdvNbx6mAa54tcZ204GmPCIhxe1UrXKd8/edit#responses

Revealed the top 3 plugins in terms of popularity are:
- Flannel
- Calico
- WeaveNet

The kubeadm maintainers are in contact with the maintainers
of these plugins and can provide support and updates to this guide.

**The rest of the plugins in the guide are unmaintained, by the kubeadm team!**

Some plugins are not in the CNCF landscape, or maintained by the kubeadm team, but still
present in the guide.

Remove the following non-CNCF / unmaintained plugins:
- AWS VPC
- Romana
- Juniper*
- Canal (it is an alternative to Calico and already explained in the
Calico guide)

Add note that users can still defer to this page for more details on Pod networking:
https://kubernetes.io/docs/concepts/cluster-administration/networking/
(the removed plugins are present there in a bigger list)

/sig cluster-lifecycle
/kind cleanup

the PR also includes some pod -> Pod renames.

xref https://github.com/kubernetes/enhancements/issues/1326

neolit123 commented:
@kubernetes/sig-cluster-lifecycle-pr-reviews 
/cc @sftim
/assign @zacharysarah 

/hold
i'm proposing that we should merge this on lazy consensus until the end of the week.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
```diff
@@ -288,13 +288,8 @@ You can install only one pod network per cluster.
 
 {{< tabs name="tabs-pod-install" >}}
 {{% tab name="Choose one..." %}}
```
Aside: I'm not sure I'd keep the “Choose one...” tab. That can come out in a different pull request.

neolit123 replied:
removed and extracted one sentence from it.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
````diff
@@ -306,16 +301,6 @@ For Calico to work correctly, you need to pass `--pod-network-cidr=192.168.0.0/1
 ```
 
-{{% /tab %}}
-{{% tab name="Canal" %}}
````
:+1:

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
```diff
@@ -383,16 +368,6 @@ For more information about `flannel`, see [the CoreOS flannel repository on GitH
 ](https://github.com/coreos/flannel).
 {{% /tab %}}
 
-{{% tab name="JuniperContrail/TungstenFabric" %}}
```
:+1:

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
```diff
@@ -405,20 +380,6 @@ Kube-router provides pod networking, network policy, and high-performing IP Virt
 For information on setting up Kubernetes cluster with Kube-router using kubeadm, please see official [setup guide](https://github.com/cloudnativelabs/kube-router/blob/master/docs/kubeadm.md).
 {{% /tab %}}
 
-{{% tab name="Romana" %}}
```
:+1:

zacharysarah commented:
@neolit123 I love this idea! I'd also love to get your feedback on the [KEP for removing third-party content](https://github.com/kubernetes/enhancements/pull/1327), given that this PR is highly relevant to that discussion.

neolit123 commented:
@sftim update:
- removed the "choose one..." tab.
- added the lnk to the networking section under "what's next"
- did some small side cleanup of "pod" -> "Pod"

neolit123 commented:
@zacharysarah 
> @neolit123 I love this idea! I'd also love to get your feedback on the KEP for removing third-party content, given that this PR is highly relevant to that discussion.

i think i already had comments on the KEP. will have another pass but overall i think it's in a good shape.

detiber commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
```diff
@@ -278,24 +278,16 @@ Make sure that your network manifest supports RBAC.
 
-You can install only one pod network per cluster.
+You can install only one Pod network per cluster.
+Bellow you can find installation instructions for some popular, CNCF approved Pod network plugins.
```
```suggestion
Below you can find installation instructions for some popular, CNCF approved Pod network plugins.
```

sftim commented:
@neolit123 I can see that it's been a while since you opened this
Would you be willing to update it incorporating upstream changes?

neolit123 commented:
@zacharysarah @sftim
i have rebased the PR on master.

/hold cancel

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md:
```diff
@@ -278,24 +278,16 @@ Make sure that your network manifest supports RBAC.
 
-You can install only one pod network per cluster.
+You can install only one Pod network per cluster.
+Below you can find installation instructions for some popular, CNCF approved Pod network plugins.
```
@neolit123 I wonder about “CNCF approved”. Is there a list elsewhere of which CNI plugins have CNCF approval?

neolit123 replied:
the list i'm referring to is the CNCF "Cloud Native Network" landscape:
https://landscape.cncf.io/category=cloud-native-network&format=card-mode&grouping=category

but technically this sentence does not include the rest of the criteria we have, and we may or may not want to include it here...

should i just change this to the following?

> Below you can find installation instructions for some popular Pod network plugins:

sftim replied:
> Below you can find installation instructions for some popular Pod network plugins:

:+1: I would, it's &hellip; less controversial.

neolit123 commented:
@sftim updated.

sftim commented:
/lgtm
based on https://github.com/kubernetes/website/pull/17694#pullrequestreview-320238876 + my judgement

and
/approve
